File: android/src/main/java/top/huic/tencent_im_plugin/message/VideoMessageNode.java
package top.huic.tencent_im_plugin.message;

import com.tencent.imsdk.v2.V2TIMManager;
import com.tencent.imsdk.v2.V2TIMMessage;
import com.tencent.imsdk.v2.V2TIMVideoElem;

import top.huic.tencent_im_plugin.message.entity.VideoMessageEntity;

/**
 * 视频消息节点
 */
public class VideoMessageNode extends AbstractMessageNode<V2TIMVideoElem, VideoMessageEntity> {
    @Override
    public V2TIMMessage getV2TIMMessage(VideoMessageEntity entity) {
        String suffix = null;
        if (entity.getVideoPath().contains(".")) {
            String[] ss = entity.getVideoPath().split("\\.");
            suffix = ss[ss.length - 1];
        }
        return V2TIMManager.getMessageManager().createVideoMessage(entity.getVideoPath(), suffix, entity.getDuration(), entity.getSnapshotPath());
    }

    @Override
    public String getNote(V2TIMVideoElem elem) {
        return "[视频]";
    }

    @Override
    public VideoMessageEntity analysis(V2TIMVideoElem elem) {
        return new VideoMessageEntity(elem);
    }

    @Override
    public Class<VideoMessageEntity> getEntityClass() {
        return VideoMessageEntity.class;
    }
}

File: ios/Classes/message/VideoMessageNode.swift
import ImSDK_Plus

//
//  TextMessageNode.swift
//  tencent_im_plugin
//
//  Created by 蒋具宏 on 2020/3/13.
//  视频消息节点
public class VideoMessageNode: AbstractMessageNode {

    override func getV2TIMMessage(params: [String: Any]) -> V2TIMMessage {
        let videoPath: String = getParam(params: params, paramKey: "videoPath")!;
        let duration: Int32 = getParam(params: params, paramKey: "duration")!;
        let snapshotPath: String = getParam(params: params, paramKey: "snapshotPath")!;

        var suffix: String = "";
        if videoPath.contains(".") {
            let ss = videoPath.split(separator: ".");
            suffix = String(ss[ss.count - 1]);
        }
        return V2TIMManager.sharedInstance().createVideoMessage(videoPath, type: suffix, duration: duration, snapshotPath: snapshotPath)
    }

    override func getNote(elem: V2TIMElem) -> String {
        "[视频]";
    }

    override func analysis(elem: V2TIMElem) -> AbstractMessageEntity {
        VideoMessageEntity(elem: elem as! V2TIMVideoElem);
    }
}
